### PR TITLE
Add interface for strict mode evaluation.

### PR DESCRIPTION
**Pull request by jpschorr**
No implementation currently, just the interface.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-conformance-tests/tests/mod.rs [hunk 1]
```diff
@@ -107,18 +117,24 @@ pub(crate) fn fail_eval(statement: &str, mode: EvaluationMode, env: &Option<Test
 
     let parsed = parse(statement);
     let lowered_result = lower(&catalog, &parsed.expect("parse"));
-    let lowered = lowered_result.expect("lower");
+    let lowered = lowered_result.expect("blah blah blah lower");
```
Comment (alancai98) [inline partiql-conformance-tests/tests/mod.rs:120]: Probably want to remove "blah blah blah"?

**Review by alancai98 — APPROVED** [review]
Looks good. Just one minor comment for the `.expect` message in conformance tests

File: partiql-conformance-tests/tests/mod.rs [hunk 2]
```diff
@@ -18,6 +18,15 @@ pub(crate) enum EvaluationMode {
     Error,
 }
 
+impl From<EvaluationMode> for eval::plan::EvaluationMode {
+    fn from(value: EvaluationMode) -> Self {
+        match value {
+            EvaluationMode::Coerce => eval::plan::EvaluationMode::Permissive,
+            EvaluationMode::Error => eval::plan::EvaluationMode::Strict,
```
Comment (am357) [inline partiql-conformance-tests/tests/mod.rs:25]: Out of the scope for this PR but should we reuse `eval::plan::EvaluationMode`?

File: partiql-conformance-tests/tests/mod.rs (hunk 1 above)
Comment (am357) [reply partiql-conformance-tests/tests/mod.rs:120]:
```suggestion
    let lowered = lowered_result.expect("lower");
```

**Review by am357 — APPROVED** [review]
LGTM w/ one inline suggestion.

**Pull request merged**